docs/cookbook/rag.md

```diff
@@ -0,0 +1,257 @@
+
+```python
+soccer_info = [
+    "Manchester City won the league, continuing their domestic dominance.",
+    "In La Liga, FC Barcelona emerged victorious with a sturdy defense.",
+    "Dortmund slipped up in the last matchday to hand Bayern the Bundesliga title.",
+    "For the first time since Maradona's days, Napoli lifted the Serie A trophy.",
+]
+```
+
+The end goal is to be able to ask our model: which team in {insert_country} won the league in the 2023 season? Then, we want to put this additional context in our prompt to obtain the correct answer.
+
+### Embeddings
+
+To pick the text most relevant to our question from our information pool, we first need to embed each of the statements in `soccer_info`. Through our wrapper class [`OpenAIChat`](../api/chat/models.md#mirascope.chat.models.OpenAIChat), we can access the `OpenAI` model through the `OpenAI.client` attribute, then call [OpenAI's embeddings](https://platform.openai.com/docs/guides/embeddings):
+
+```python
+from mirascope import OpenAIChat, Prompt
+
+chat = OpenAIChat(api_key="YOUR_OPENAI_API_KEY")
+embeddings_model = "text-embedding-ada-002"
+
+embeddings = [
+    chat.client.embeddings.create(model=embeddings_model, input=[text])
+    .data[0]
+    .embedding
+    for text in soccer_info
+]
+
+df = pd.DataFrame({"texts": soccer_info, "embeddings": embeddings})
+```
+
+### Retrieval
+
+Here is the `Prompt` we will be using to ask our model:
+
+```python
+class SoccerPrompt(Prompt):
+    """
+    Here is some context about what happened in soccer in the 2022-23 season:
+    {context}
+
+
+    Based on this information, answer the following question:
+    {question}
+    """
+
+    context: str
+    question: str
+```
+
+But to provide `context`, we need to implement the retrieval functionality. To do this, we embed our question (`query_embedding`), then rank the similarities between `query_embedding` and each of the embeddings from `soccer_info` using the dot product between the vectors. The text with the highest value (and thus, similarity) is passed in as the context. And just like that, our basic RAG example is complete.
+
+```python
+def ask_soccer(query):
+    """Answers a question about soccer from retrieved context."""
+    query_embedding = (
+        chat.client.embeddings.create(model=embeddings_model, input=[query])
+        .data[0]
+        .embedding
+    )
+    # Embedded texts are all the same length, so dot is equivalent to cosine
+    df["similarities"] = df.embeddings.apply(lambda x: np.dot(x, query_embedding))
+    most_similar = df.sort_values("similarities", ascending=False).iloc[0]["texts"]
+
+    prompt = SoccerPrompt(context=most_similar, question=query)
+    completion = chat.create(prompt)
+    print(completion)
+
+
+countries = ["English", "Spanish", "German", "Italian"]
+for country in countries:
+    ask_soccer(f"Who won the {country} top flight?")
+```
+
```

Comment:
add short blurb to showcase how the soccer_info above matches the print statement